### ISSUE TITLE
Bulk Upload - Pulls off email attachments twice

### ISSUE THREAD

**Issue by gsteimer** [tsgrp/HPI#1336]
If bulk upload is run twice in a row without refreshing in between, email attachments are pulled off twice. 
 This is causing STS to fail.  Here's the scenario:

1. Run bulk upload to upload a document.  Should work fine.
1. _Without refreshing_, run bulk upload again and select an MSG file with attachments.  The email attachments will be pulled off twice.  See the screencam below.

In the below screencam I'm in a Policy folder in the Insurance trac on edge2.  I've just uploaded a single document.  Note - you can find the `Quote for Public DNO.msg` file in `HPI\trunk\demo-docs\Insurance`.  This email has two PDF attachments.

![attachment-2x](https://cloud.githubusercontent.com/assets/1377228/24167060/6657de06-0e43-11e7-9b62-4c32b2c71f9f.gif)

**Comment by gsteimer** [comment]
Note - this may be related to #1309

**Comment by Kovacinski** [comment]
Fixed with revision: 17629

**Comment by joehof** [comment]
After talking with with Steve, I want to re-open this. The fix that was put in was really a band-aid (the view itself was looking for any other views of itself and removing them). The onus of cleaning up this view should be withe the parent view, not itself.

**Comment by gsteimer** [comment]
@joehof - can we keep this one closed and open up a new issue for the clean up?

**Comment by gsteimer** [comment]
@joehof @Kovacinski - scratch closing this issue.  It's still happening on edge2 and STS scripts are failing:

![image](https://cloud.githubusercontent.com/assets/1377228/24468928/6ccbab18-147f-11e7-84a4-947b80370644.png)

**Comment by gsteimer** [comment]
I'm also seeing issues on alfresco with normal document uploads.  For example:

1. Upload document A
1. Without refreshing, run Bulk Upload again and upload Document B
1. It will fail with the following error:

![image](https://cloud.githubusercontent.com/assets/1377228/25594481/5fb3e85c-2e86-11e7-8209-f69fc3d1c558.png)

```
Caused by: org.springframework.dao.DuplicateKeyException: 
### Error updating database.  Cause: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "parent_node_id"
  Detail: Key (parent_node_id, type_qname_id, child_node_name_crc, child_node_name)=(4373, 33, 3503519332, document import - mon, 1 may 2017 03-53-21 pm) already exists.
### The error may exist in alfresco/ibatis/#resource.dialect#/node-insert-SqlMap.xml
### The error may involve alfresco.node.insert.insert_ChildAssoc-Inline
### The error occurred while setting parameters
### SQL: insert into alf_child_assoc         (             id, version, parent_node_id, child_node_id, type_qname_id,             child_node_name_crc, child_node_name,             qname_ns_id, qname_localname, qname_crc,             is_primary, assoc_index         )         values         (             ?, ?, ?, ?, ?,             ?, ?,             ?, ?, ?,             ?, ?         )
### Cause: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "parent_node_id"
  Detail: Key (parent_node_id, type_qname_id, child_node_name_crc, child_node_name)=(4373, 33, 3503519332, document import - mon, 1 may 2017 03-53-21 pm) already exists.
; SQL []; ERROR: duplicate key value violates unique constraint "parent_node_id"
  Detail: Key (parent_node_id, type_qname_id, child_node_name_crc, child_node_name)=(4373, 33, 3503519332, document import - mon, 1 may 2017 03-53-21 pm) already exists.; nested exception is org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "parent_node_id"
  Detail: Key (parent_node_id, type_qname_id, child_node_name_crc, child_node_name)=(4373, 33, 3503519332, document import - mon, 1 may 2017 03-53-21 pm) already exists.
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:245)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:72)
	at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:74)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:399)
	at com.sun.proxy.$Proxy11.insert(Unknown Source)
	at org.mybatis.spring.SqlSessionTemplate.insert(SqlSessionTemplate.java:253)
	at org.alfresco.repo.domain.node.ibatis.NodeDAOImpl.insertChildAssoc(NodeDAOImpl.java:932)
	at org.alfresco.repo.domain.node.AbstractNodeDAOImpl.newChildAssocInsertImpl(AbstractNodeDAOImpl.java:3236)
	... 133 more
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "parent_node_id"
  Detail: Key (parent_node_id, type_qname_id, child_node_name_crc, child_node_name)=(4373, 33, 3503519332, document import - mon, 1 may 2017 03-53-21 pm) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2270)

```

The odd thing is, if I refresh the folder, it seems like the both documents were uploaded properly

**Comment by grahamsinger** [comment]
Update on this (and possibly #1384):

Using Chrome, I was able to replicate the "seeing 2 emails" on `edge2` maybe 3 times total today.  I wasn't able to replicate locally using `grunt server` or `grunt server:debug` (not uglified).

Interestingly, when using Safari, I am able to replicate consistently on `edge2`, but again, not once locally.

In the scenario I'm testing with, the listeners don't get destroyed (in Safari) when doing either

- a `removeView()` on the old view, or
- a `setView()` with a new view

(both eventually call `stopListening()`, which calls `off()` which is what should be removing that event listener).  

In short, @Kovacinski's fix is both working and failing for me in certain environment's consistently.  It's also behaving inconsistently on edge2 using Chrome, which might be a sign that there is more going wrong (as @m-brouillette mentioned [here)](https://github.com/tsgrp/HPI/issues/1384#issuecomment-299026150)

If anyone has ideas or can find an iron-clad replicable error scenario on chrome, please let me know!

**Comment by grahamsinger** [comment]
Closed with rev17854
